### DISCUSSION TITLE
Making a binder

### DISCUSSION THREAD

**Discussion by marisalim** [nih-cfde/training-and-engagement#364] [Q&A] [answered]
How do I make a binder like the one used in the Intro to Conda workshop?

**Comment by marisalim** [comment, marked as answer]
We used a Pangeo Binder for the workshop: https://binder.pangeo.io/

To make a binder, you need to create a [Github account](https://github.com/) and a Github repository to put files you want to use for the binder.

The repo should have a folder called `binder`. In this folder, you add files that specify the software environment for the binder. For example, the workshop's binder used this `environment.yml` file (with conda!) to specify installing R so we could use the Rstudio interface:

```
channels:
    - conda-forge
    - bioconda
    - defaults
dependencies:
    - r-base
```

After adding all the files we want in the binder, go to the pangeo website and enter the Github repo URL. Optional: add the repo branch you're working on if not the `main` branch to point pangeo to. If you want an Rstudio interface, change "Path to a notebook file (optional)" dropdown to "URL" instead of "File". Type `rstudio` so the binder opens Rstudio. Then click "launch" and wait for the binder to build. Copy the binder badge text to create a clickable button.

![](https://i.imgur.com/v1lwGJ8.png)

Here are some examples for rstudio binders: https://github.com/binder-examples/r-conda

You can also make binders that open with jupyter notebooks or terminal interfaces!

We plan to add a more detailed lesson about making binders to our training website in the near future!

**Reply by ctb** [reply]
The full guide to what files to put where for configuring a binder is here - 

https://repo2docker.readthedocs.io/en/latest/config_files.html

-- repo2docker is what builds the docker containers used by binder.
